packages/salesforcedx-vscode-soql/src/queryDataView/queryDataViewService.ts

```diff
@@ -113,28 +117,35 @@ export class QueryDataViewService {
     return this.currentPanel.webview;
   }
 
-  protected onDidRecieveMessageHandler(message: DataViewEvent) {
+  protected onDidRecieveMessageHandler(message: DataViewEvent): void {
     const { type, format } = message;
     switch (type) {
       case 'activate':
         this.updateWebviewWith(this.queryData);
         break;
       case 'save_records':
-        this.handleSaveRecords(format!);
+        this.handleSaveRecords(format as FileFormat);
         break;
       default:
-        console.log('unknown message type from data view');
+        showAndTrackError(
+          'data_view_message_type',
+          `Error: ${type}. This is an unknown error.  Open an issue: https://github.com/forcedotcom/soql-tooling/issues/new/choose.`
+        );
         break;
     }
   }
 
-  protected handleSaveRecords(format: FileFormat) {
-    const fileService = new FileService(
-      this.queryData,
-      format,
-      getDocumentName(this.document)
-    );
-    fileService.save();
+  protected handleSaveRecords(format: FileFormat): void {
+    try {
+      const fileService = new FileService(
+        this.queryData,
+        format,
+        getDocumentName(this.document)
+      );
+      fileService.save();
+    } catch (err) {
+      trackError('data_view_save', err);
```

Comment:
This could include username info, it should be cleaned/removed before sending it to the telemetry service